### PR TITLE
Resolves issue 811 documentation

### PR DESCRIPTION
**Pull request by rockyallen**

### CODE REVIEW

**Comment by robertpanzer** [comment]
Thank you for your pull request!
The file docs/integrator-guide.adoc is actually generated during build from asciidoctorj-documentation/src/main/asciidoc/integrator-guide.adoc.
Could you please move your updates to that document, run the build and then add both files to your PR?

File: docs/integrator-guide.adoc [hunk 1]
```diff
@@ -1577,6 +1626,19 @@ org.asciidoctor.integrationguide.extension.TerminalCommandExtension
...
+
+[NOTE]
+--
+If you have installed AsciidoctorJ as recommended, asciidoctorj will be on the path, and you can use:
+
+ asciidoctorj -cp=lib/myextension.jar test.adoc
+
+If you have downloaded the distribution jars only, use a command like:
+
+ java -cp lib/asciidoctor-api-2.0.0.jar;lib/asciidoctor-core-2.0.0.jar;lib/jcommander-1.35.jar;lib/myextension.jar  org.asciidoctor.jruby.cli.AsciidoctorInvoker  test.adoc
```
Comment (robertpanzer) [inline docs/integrator-guide.adoc]: I guess this won't work as the jar of jruby is missing on the classpath

File: docs/integrator-guide.adoc [hunk 2]
```diff
@@ -1539,7 +1588,7 @@ To make AsciidoctorJ use our processor it also has to be registered at the `Java
 <4> Test via the Jsoup HTML parsing library that our meta tag was correctly added to the resulting document.
 
 [[ExtensionSPI]]
-=== Extension SPI
+===Automatically loading extensions
```
Comment (robertpanzer) [inline docs/integrator-guide.adoc]: Can you please add a blank between the `===` and `Automatically`?
Asciidoctor won't recognise it as a heading like this.

File: docs/integrator-guide.adoc [hunk 3]
```diff
@@ -556,35 +554,64 @@ AsciidoctorJ brings this extension API to the JVM environment.
...
+
+|<<postprocessor,Postprocessor>>
 |org.asciidoctor.extension.Postprocessor
+|Modifies the backend-specific output document
 
-|Block processor
-|org.asciidoctor.extension.BlockProcessor
+|<<docinfoprocessor,DocinfoProcessor>>
+|org.asciidoctor.extension.DocinfoProcessor
+|Insert content into the `header` element (html) or `docinfo` element (docbook)
```
Comment (robertpanzer) [inline docs/integrator-guide.adoc]: Should we also mention that it also inserts content in the footer?

File: docs/integrator-guide.adoc (hunk 1 above)
Comment (robertpanzer) [inline docs/integrator-guide.adoc]: It looks like the 2 documents are not in sync, the "source" contains the jruby.jar on the classpath, this doesn't. 
Can you rerun the build please?

**Comment by rockyallen** [comment]
Is there a special command to do this? I tried 
./gradlew clean
then
./gradlew
It built for 13 minutes with no obvious errors, but the doc/ files did not change.
Out of interest, why is it necessary to store intermediate build files in the repository?

**Comment by robertpanzer** [comment]
`./gradlew asciidoctorj-documentation:build` should update the document in `docs/`.

I went this way, because I want to have tested code snippets in this document, therefore I used includes to the test sources.
But as Github doesn't support includes I added that extra step to the build that resolves them and stores the resolved document in `docs/`,

**Comment by robertpanzer** [comment]
Otherwise you might want to explicitly trigger this task: https://github.com/asciidoctor/asciidoctorj/blob/master/asciidoctorj-documentation/build.gradle#L13
But usually `./gradlew build` does the job for me.

**Comment by robertpanzer** [comment]
Thank you!

**Pull request merged**